vscode/src/gh-copilot/instructions.ts

```diff
@@ -4,295 +4,361 @@
-  const workspaceFolder = vscode.workspace.getWorkspaceFolder(resource)?.uri;
-  if (!workspaceFolder) {
-    return;
-  }
+    let text = "";
+    try {
+      const content = await vscode.workspace.fs.readFile(instructionsFile);
+      text = new TextDecoder("utf-8").decode(content);
+      const startIndex = text.indexOf(oldCodingInstructionsTitle);
+      if (startIndex === -1) {
+        continue;
+      }
+      let endIndex = text.indexOf(oldCodingInstructionsFooter, startIndex);
+
+      if (endIndex !== -1) {
+        endIndex += oldCodingInstructionsFooter.length;
+        // Skip any trailing newlines after the footer
+        while (
+          endIndex < text.length &&
+          (text[endIndex] === "\n" || text[endIndex] === "\r")
+        ) {
+          endIndex++;
+        }
 
-  if (await hasQSharpCopilotInstructions(workspaceFolder)) {
-    // If the file already exists and contains Q# instructions, do nothing
-    return;
+        // Create new content without the Q# instructions
+        const newContent =
+          text.substring(0, startIndex) + text.substring(endIndex);
+
+        // Write back the file without the Q# instructions
+        await vscode.workspace.fs.writeFile(
+          instructionsFile,
+          new TextEncoder().encode(newContent),
+        );
+      }
+      removed = true;
+    } catch {
+      // file doesn't exist or we couldn't edit it
+      continue;
+    }
   }
 
-  sendTelemetryEvent(EventType.UpdateCopilotInstructionsStart, {}, {});
+  return removed;
+}
 
-  // Show a yes/no prompt to the user
-  const response = await vscode.window.showInformationMessage(
-    "We're about to update your `copilot-instructions.md` file.\n\n" +
-      "This file helps GitHub Copilot understand and work better with Q# files and features provided by the Quantum Development Kit extension.\n\n" +
-      "Would you like to proceed with updating `copilot-instructions.md`?",
-    { modal: true },
-    { title: "Yes" },
-    { title: "No", isCloseAffordance: true },
+function getOldInstructionsFileLocation(
+  workspaceFolder: vscode.Uri,
+): vscode.Uri {
+  return vscode.Uri.joinPath(
+    workspaceFolder,
+    ".github",
+    "copilot-instructions.md",
   );
+}
 
-  if (response?.title !== "Yes") {
-    sendTelemetryEvent(EventType.UpdateCopilotInstructionsEnd, {
-      reason: "User canceled",
-      flowStatus: UserFlowStatus.Aborted,
-    });
-    return; // User canceled or dismissed the dialog
-  }
-
-  try {
-    const githubDir = vscode.Uri.joinPath(workspaceFolder, ".github");
-    const instructionsFile = vscode.Uri.joinPath(
-      githubDir,
-      "copilot-instructions.md",
-    );
-
-    // Create .github directory if doesn't exist
-    await vscode.workspace.fs.createDirectory(githubDir);
+async function copyInstructionsFileToGlobalStorage(
+  context: vscode.ExtensionContext,
+) {
+  const source = vscode.Uri.joinPath(
+    context.extensionUri,
+    "chat-instructions",
```

Comment:
Should the `chat-instructions` dir go inside the `resources` dir to keep the top-level `vscode` dir a little cleaner, and consolidate where the extension resources go?

Reply:
Done